### ISSUE TITLE
Request: translator and history import

### ISSUE THREAD

**Issue by Wowozell** [agnaistic/agnai#118]
it would be nice to have automatic two-way translation for talking in different languages.  this thing seems to be simple, but pleasant for those many for whom English is not native. It would also be nice to be able to import dialogue histories with characters in any form.

**Comment by jonasnordlund** [comment]
I agree with this and think this would be a fairly unique feature among services of this kind. I've requested it elsewhere myself but I've never seen it in action.

My idea is a [Foreign] -> [English] -> LLM -> [English] -> [Foreign] flow that integrates with a third-party translation service, so the LLM itself would never need to understand or speak anything other than English but that this would be transparent to the user in action.

The added cost could be handled by the third-party translation service which the user has an API key and is charged for separately, so that it is not falling onto Agnaistic.

The entire chat history/context window, prompts etc, it would all remain in English and this would be a purely user-interface level feature where only each new snippet that is being exchanged is sent for translation.

Still, my remaining concern is that the translators of today might not preserve subtle tone and wording etc. intact. I'm not sure if there are any modern, cost-effective translators up to par here, also offering an API that isn't too costly or e.g. only geared for enterprise use?
